src/type.ts

```diff
@@ -335,10 +335,10 @@ export class Type {
 	}
 
 	get properties(): Property[] {
-		let propertiesArray: Property[] = [];
-		for (var type: Type = this; type != null; type = type.baseType) {
+		let propertiesArray: Property[] = Object.values(this.__properties__);
+		for (var type: Type = this.baseType; type != null; type = type.baseType) {
 			for (var propertyName in type.__properties__) {
-				if (type.__properties__.hasOwnProperty(propertyName)) {
+				if (!this.__properties__.hasOwnProperty(propertyName)) {
```

Comment:
It seems more correct to check a local map of the properties being gathered instead of `this.__properties__`. Right now, if there was a type hierarchy 3 levels deep, and the second level overrode a property from the top level, you would end up with both properties in the bottom type's list of `properties`.